pkg/vault/test/foundry/ProtocolFeePercentagesProvider.t.sol

```diff
@@ -0,0 +1,285 @@
+
+    IAuthentication internal percentagesProviderAuth;
+    IAuthentication internal feeControllerAuth;
+
+    uint256 internal maxSwapFeePercentage;
+    uint256 internal maxYieldFeePercentage;
+
+    address[] internal pools;
+
+    function setUp() public override {
+        BaseVaultTest.setUp();
+
+        trustedContractRegistry = new BalancerContractRegistry(vault);
+        percentagesProvider = new ProtocolFeePercentagesProvider(vault, feeController, trustedContractRegistry);
+
+        // Mark the poolFactory as trusted, so that operations on it won't fail.
+        authorizer.grantRole(
+            trustedContractRegistry.getActionId(BalancerContractRegistry.registerBalancerContract.selector),
+            admin
+        );
+        authorizer.grantRole(
+            trustedContractRegistry.getActionId(BalancerContractRegistry.deprecateBalancerContract.selector),
+            admin
+        );
+        vm.prank(admin);
+        trustedContractRegistry.registerBalancerContract(ContractType.POOL_FACTORY, "MockFactory", poolFactory);
+
+        percentagesProviderAuth = IAuthentication(address(percentagesProvider));
+        feeControllerAuth = IAuthentication(address(feeController));
+
+        (maxSwapFeePercentage, maxYieldFeePercentage) = feeController.getMaximumProtocolFeePercentages();
+
+        // Ensure we aren't comparing to 0.
+        require(maxSwapFeePercentage > 0, "Zero swap fee percentage");
+        require(maxYieldFeePercentage > 0, "Zero yield fee percentage");
+
+        pools = new address[](1);
+        pools[0] = pool;
+    }
+
+    function testInvalidConstruction() public {
+        vm.expectRevert(IProtocolFeePercentagesProvider.WrongProtocolFeeControllerDeployment.selector);
+        new ProtocolFeePercentagesProvider(IVault(INVALID_ADDRESS), feeController, trustedContractRegistry);
+    }
+
+    function testGetProtocolFeeController() public view {
+        assertEq(
+            address(percentagesProvider.getProtocolFeeController()),
+            address(feeController),
+            "Wrong protocol fee controller"
+        );
+    }
+
+    function testGetFactorySpecificProtocolFeePercentagesUnregisteredFactory() public {
+        vm.expectRevert(
+            abi.encodeWithSelector(IProtocolFeePercentagesProvider.FactoryFeesNotSet.selector, INVALID_ADDRESS)
+        );
+        percentagesProvider.getFactorySpecificProtocolFeePercentages(INVALID_ADDRESS);
+    }
+
+    function testSetFactorySpecificProtocolFeePercentageNoPermission() public {
+        vm.expectRevert(IAuthentication.SenderNotAllowed.selector);
+        percentagesProvider.setFactorySpecificProtocolFeePercentages(
+            poolFactory,
+            maxSwapFeePercentage,
+            maxYieldFeePercentage
+        );
+    }
+
+    function testRevertWhenSetFactorySpecificProtocolFeePercentageInvalidFactory() public {
+        _grantPermissions();
+
+        vm.expectRevert();
```

Comment:
Let's be specific with the revert reasons in the tests.

Reply:
Not sure why I didn't do that. Will add this to all of them.